mtupper/homework.md

```diff
@@ -0,0 +1,292 @@
+
+**c)** `x == 'x'`<br />
+> **false** - the only way to make this true would be to declare a variable x with a value of the string 'x'<br />
+> `var x = "x"`<br />
+
+**d)** `x == (x+'')`<br />
+> **true** - since adding nothing (empty string `''`) to the variable x won't change its value<br />
+
+**e)** `'' == ' '`<br />
+> **false** - because an empty string is not the same as a string consisting of a space character<br />
+
+**f)** `x = true`<br />
+> **true** - since we just set the value of x equal to true, it is indeed true<br />
+
+**g)** `var x; x == 'undefined'`
+> **false** - while it was not assigned a value, the variable x has been defined, same as assigning value `''`<br />
+
+**h)** `'9'<'10'`<br />
+> **false** - because they are both a `"string"`, strings are ordered alphabetically (1 before 9) and therefore, 10 comes before 9<br />
+
+**i)** `typeof x + 1 === "number"`<br />
+> **false** - ```typeof x + 1``` returns "number1", adding 1 to the result of typeof x<br />
+
+**j)** `typeof x % 2 === "number"`<br />
+> **false** - ```typeof x % 2``` returns NaN because it is the same as ```"number" % 2```<br />
+
+**k)** `typeof (x % 2) === "number"` <br />
+> **true** - makes sense when x is already a number
+>
+> ```javascript
+> var x = 4;
+> typeof (x % 2); // returns "number"
+> ```
+> However, I am not clear why this `typeof` returns `"number"`
+>```javascript
+> var x = "text";
+> (x % 2); // returns NaN, but...
+> typeof (x % 2); // returns "number"
+>```
+
+**l)** `x++ == ++x`
+> **false** - value of x is returned and then incremented and incremented again before being compared to the new value of x
+>
+> ```javascript
+> var x = 1;
+> var y = x++; // y = 1, x = 2
+> var z = ++x; // z = 3, x = 3
+> y == z; // is false
+> ```
+
+**m)** `++x == x++`
+> **true** - value of x is incremented and returned, new value of x is compared to x (true), then incremented.
+>
+> ```javascript
+> var x = 1;
+> var y = ++x; // y = 2, x = 2;
+> var z = x++; // z = 2, x = 3;
+> y == z; // is true
+> ```
+
+**n)** `"1"+x == 1+x`
+> **false** - treats addition of strings different from addition of integers
+>
+> ```javascript
+> var x = 4;
+> "1" + x; // returns "14"
+>
+> var x = 4;
+> 1 + x; // returns 5
+>
+> "14" == 5; // returns false
+> ```
+
+**o)** `"0"+1 == 1`
+> **true** - will convert strings to numbers for comparison with non-strict (==) equality
+> ```javascript
+> "0" + 1; // returns "01"
+>
+> "01" == 1; // returns true
+> ```
+
+**p)** `(typeof (x+1))===(typeof x)`
+> **true** - comparing `"numbers"`
+> ```javascript
+> var x = 4;
+> (typeof (x + 1)); // is same as (5) and returns "number"
+> (typeof x); // is same as 4 and also returns "number"
+> ```
+
+**q)** `(x*1 == x) || ((typeof x) != "number")`
+> **true** - since `(x*1 == x)` evaluates to true, second statement is optional (or `||`), it returns true without needing to evaluate the second statement
+
```

Comment:
Not quite: first clause (x*1 == x) fails for non-numeric strings, undefined, and NaN.  The second clause makes it true for strings and undefined, but NaN still fails both.